src/coreclr/src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -422,114 +422,15 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 void CodeGen::genHWIntrinsic_R_RM(
     GenTreeHWIntrinsic* node, instruction ins, emitAttr attr, regNumber reg, GenTree* rmOp)
 {
-    emitter* emit = GetEmitter();
-
     assert(reg != REG_NA);
+    assert(!node->OperIsCommutative()); // One operand intrinsics cannot be commutative
```

Comment:
The method header doesn't indicate that the first "operand" must be a destination, so it's not clear that this is always a one-operand node. The header should be changed, and the argument should probably be changed from `reg` to `targetReg`.